libs/SalesforceSDK/src/com/salesforce/androidsdk/config/RuntimeConfig.java

```diff
@@ -78,6 +80,10 @@ private RuntimeConfig(Context ctx) {
 				SalesforceSDKManager.getInstance().registerUsedAppFeature(FEATURE_MDM);
 			}
 
+			if(getBoolean(RuntimeConfig.ConfigKey.RequireCertAuth)){
```

Comment:
Move this inside the `if(isManaged && configurations!=null && !configurations.isEmpty())` block.